### PR TITLE
test(cat): gate keyerless-dependent CW tests (TS-2000 §12.13, rigctld E6)

### LINKED ISSUE #4855: test(cat): keyerless CW tests run ungated on keyerless rigs — TS-2000 §12.13 KY;, rigctld E6 send_morse
### Summary
The TS-2000 CAT integration test asserts `KY;` returns `KY0`/`KY1` at **§12.13**, inside the *ungated* Tier-2 `section12`. On any radio without a radio-side CW keyer — including the built-in demo (`SimBackend`) — `SmartCatProtocol::cmdKY` correctly answers `?;` (it gates on `hasRadioSideCwKeyer()`), so the assertion spuriously fails on a normal no-TX / demo run.

The same class of bug exists in the rigctld suite: **E6** runs `send_morse "TEST"` unconditionally in the ungated Edge-Cases block. `cmdSendMorse` returns `RPRT -11` (`RIG_ENAVAIL`) on a keyerless rig, so E6 also fails on the demo — and, worse, `send_morse` routes → `CwxModel` → `"cwx"`, which **keys CW on a real radio**, so a plain no-TX run transmits on the operator's licence (Principle VI). FlexCAT is already correct — its `KY` tests live in the `--cw`-gated `section14`. So 2 of the 3 suites need gating.

### Evidence
Against the built-in demo, the TS-2000 suite is **95/96**, the sole failure being:
```
FAIL  12.13 KY; (query) → KY0 (not busy) or KY1 (busy)  → "?"
```
and rigctld E6 (`send_morse`) fails identically with `RPRT -11` on a no-`--cw` run. `12.13` is byte-identical on `main`, so it fails there too.

### Root cause
`KY` / `send_morse` are CW-keyer capabilities. They only behave meaningfully on a rig that *has* a radio-side keyer — exactly the precondition the suite already models with the `--cw` flag. `section13` (TS-2000 CW-keyer tests) is already `--cw`-gated and, in fact, **already queries `KY;` as its `13.1` baseline**; §11 (rigctld CW/PTT) is already gated too. §12.13 and E6 exercise keyer commands but sit in the *ungated* sections, so they run (and fail) on keyerless / demo targets.

### Fix
Gate the keyerless-dependent checks behind `--cw`, mirroring the gating already established for the CW/PTT tests:

- **TS-2000 §12.13** — gate the `KY;` query behind `--cw` (thread `doCw` into `section12`; run the check under `--cw`, `r.skip()` it otherwise), mirroring the suite's `section13skip`. Keeps the Tier-2 command visible as a SKIP on ungated runs rather than silently dropping it, while keyered rigs keep the coverage. `KY` stays listed in section 12's command strings. (The `KY;` query is read-only — it reports keyer-busy state, `KY<text>;` is what keys — so it was never a transmit hazard; only its ungated assertion was wrong.) The redundancy vs §13.1 is kept deliberately: §12.13 is the Tier-2 **presence** check (`KY0` or `KY1`), §13.1 is the pre-send **state** baseline (`== "KY0"`).
- **rigctld E6** — gate `send_morse "TEST"` behind `--cw` (`sectionEdge` takes `doCw`, skips with a reason otherwise), exactly like §11. This is the safety-relevant half: it stops an ungated suite run from keying a transmitter.

A no-TX / demo run then skips these keyer tests — no spurious failure, no keying — while a real `--cw` run on a keyer-equipped rig still verifies them.

### Repro
Run `CAT_TS-2000_test --port <ts2000-port>` (or `rigctld_test --port <port>`) against the built-in demo **without** `--cw` → §12.13 / E6 fail.

---
🤖 Reported with assistance from [Claude Code](https://claude.com/claude-code)

### PR DESCRIPTION
**Pull request by K5PTB**
Fixes #4855.

Two CAT integration tests asserted a CW-keyer command succeeds unconditionally, but those commands only work on a rig with a *radio-side* keyer — a keyerless target (including the built-in demo / `SimBackend`) correctly refuses them, so the tests fail on every no-TX / demo run, and E6 would key CW on real hardware. The fix mirrors the suite's existing `--cw` / `--ptt` safety gating.

## Changes (one test file each, no production code)

**`CAT_TS-2000_test.cpp` — gate the §12.13 `KY;` query behind `--cw`.** `section12` (ungated Tier-2) asserted `KY;` → `KY0`/`KY1`, but `SmartCatProtocol::cmdKY` answers `?;` when `hasRadioSideCwKeyer()` is false, so it failed on every keyerless / demo run. §12.13 now runs under `if (doCw)` and `r.skip()`s otherwise (`doCw` threaded into `section12`), mirroring the suite's own `section13skip` convention — the check stays visible as a SKIP on ungated runs while keyered rigs (any Flex) keep the coverage. `KY` remains listed in section 12's two command-list strings. Note the `KY;` **query** is read-only (it reports keyer-busy state; `KY<text>;` is the form that keys), so unlike E6 it was never a transmit hazard — only its ungated `== KY0/KY1` assertion was wrong.

> **Note (updated after review):** an earlier revision of this body said §12.13 was *removed* and `KY` *dropped* from the section header. That's not what shipped — it's **gated**, and `KY` stays in the header. Gating keeps the Tier-2 command visible (a Tier-2 section that silently loses a Tier-2 command is worse than one that skips it). The redundancy #4855 flagged therefore survives, deliberately: on a keyered `--cw` run `KY;` is queried twice — §12.13 is the Tier-2 **presence** check (accepts `KY0` **or** `KY1`), §13.1 is the pre-send **state** baseline (stricter `== "KY0"`). Distinct roles, kept on purpose.

**`rigctld_test.cpp` — gate E6 `send_morse` behind `--cw`.** E6 ran `send_morse "TEST"` in the ungated Edge-Cases block asserting `RPRT 0`, but `cmdSendMorse` routes `send_morse` → `CwxModel` → `"cwx"`, which returns `RPRT -11` (`RIG_ENAVAIL`) on a keyerless rig **and keys CW on a real radio** — so a plain no-TX run transmitted on the operator's licence. `sectionEdge` now takes `doCw` and skips E6 with a clear reason when `--cw` isn't set, exactly like §11. This is the safety half of the fix.

**FlexCAT — no change needed.** Its `KY` tests already live in the `--cw`-gated `section14`.

## Testing (built-in demo, no `--cw`, Mac)
- **TS-2000:** 89/89, 0 failed, §12.13 `SKIP`s (was 89/90 — the one failure being §12.13). Verified live against the sim.
- **rigctld:** E6 now `SKIP`s — no keying, no spurious failure. (The suite's §5 split failures against the single-slice demo are the separate `--no-split` matter, untouched here.)

Built and smoke tested on: MacBook Air M2 (test-only change)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/CAT_TS-2000_test.cpp [hunk 1]
```diff
@@ -807,10 +807,10 @@ void section12(CatClient& c, Runner& r)
...
     if (origPt.startsWith(QLatin1String("PT"))) { c.send(origPt); QThread::msleep(50); }
 
-    // ── KY: CW send / busy query ─────────────────────────────────────────────
-    resp = c.query(QStringLiteral("KY"));
-    r.check(QStringLiteral("12.13 KY; (query) → KY0 (not busy) or KY1 (busy)"),
-            resp == QLatin1String("KY0") || resp == QLatin1String("KY1"), repr(resp));
+    // KY (CW keyer busy query) is exercised in section 13, which is --cw-gated:
+    // it only answers KY0/KY1 on a rig with a radio-side keyer, so a keyerless
+    // target (e.g. the built-in demo) correctly returns "?;". It has no place in
+    // this ungated Tier-2 section.
```
[the remark this answers is not in the document]
Comment (K5PTB) [reply tests/CAT_TS-2000_test.cpp]: Took the gate-with-skip route in `4cfc47b0` — you're right that the `KY;` query is read-only (it reports keyer-busy state; `KY<text>;` is the form that keys), so the only defect was the ungated `== KY0/KY1` assertion, not the check existing. §12.13 now runs under `--cw` and `r.skip()`s otherwise, mirroring `section13skip`, so it stays visible as a SKIP on keyerless/ungated runs and keyered rigs (any Flex) keep the coverage. `doCw` is threaded into `section12`, and `KY` is back in the two section-12 command-list headers. Both files now consistently gate-not-delete.

File: tests/rigctld_test.cpp [hunk 2]
```diff
@@ -2074,12 +2074,18 @@ void sectionEdge(RigctlClient& c, Runner& r)
...
-            c.ok(lines), lines.join(QStringLiteral(" | ")));
-    c.send(QStringLiteral("\\stop_morse"));
+    // E6  send_morse smoke test — verify the protocol accepts/queues the command.
+    //     GATED behind --cw: send_morse routes through CwxModel → "cwx", which
+    //     KEYS CW on real hardware, so it must not run on a plain (no-TX) pass.
+    if (doCw) {
+        lines = c.send(QStringLiteral("\\send_morse TEST"));
+        r.check(QStringLiteral("E6  send_morse \"TEST\" accepted (RPRT 0)"),
+                c.ok(lines), lines.join(QStringLiteral(" | ")));
+        c.send(QStringLiteral("\\stop_morse"));
```
[the remark this answers is not in the document]
Comment (K5PTB) [reply tests/rigctld_test.cpp:2084]: Agreed it's now effectively a duplicate of §11.1 under `--cw`. Keeping the gated copy for the reasons you noted (cheap, and E-numbering stability has value), and with §12.13 now gated rather than deleted the two files take the same route. Thanks.

**Review by ten9876 — COMMENTED** [review]
## Issue fit — #4855

**Issue:** two CAT integration tests assert a CW-keyer command succeeds unconditionally. On a keyerless target — including the built-in demo / `SimBackend` — `cmdKY` answers `?;` and `cmdSendMorse` returns `RPRT -11`, so both fail on every no-TX/demo run; and on real hardware `send_morse` **keys CW**.

**Does it solve it? Yes for the safety half, which is the half that matters.** Gating E6 behind `--cw` stops an ungated suite run from keying a transmitter, and that's squarely Principle VI territory — *"it never causes a transmission the operator did not deliberately initiate"*. `sectionEdge(c, r, doCw)` mirrors §11's existing gating, `doCw` is already parsed at `rigctld_test.cpp:2278` and threaded correctly, and the skip carries a reason a reader can act on.

The TS-2000 half solves the *failure*, but not by the route the issue or your own body describes — see below.

## Blockers

None. Nothing here is unsafe or wrong; both items are accuracy.

## The PR body doesn't match the diff

Your summary says:

> **`CAT_TS-2000_test.cpp` — remove the redundant §12.13 `KY;` query.** […] §12.13 was a duplicate on the wrong side of the gate — **removed, and `KY` dropped from section 12's command list.**

and #4855's Fix section asks for the same thing (*"Consolidate the `KY;` query into the `--cw`-gated `section13` […] and remove the duplicate §12.13"*).

The diff does neither. §12.13 is **gated**, not removed — the assertion is still there under `if (doCw)`, with an `else` that skips. And `KY` is still in section 12's command list in both places it appears: the banner comment at `:723` and the `r.section()` string at `:728`, both unchanged.

I actually think **gating is the better call** than removing, and your own inline comment makes the case (*"while keyered rigs (any Flex) keep the coverage"*) — a Tier-2 section that silently loses a Tier-2 command is worse than one that skips it. So this is a request to fix the *description*, not the code.

But it has a consequence worth deciding on: the redundancy #4855 identified now **survives**. Under `--cw` on a keyered rig, `KY;` is queried twice — §12.13 (`:818`, accepts `KY0` **or** `KY1`) and §13.1 (`:904`, the stricter `== "KY0"` baseline). Two options, either fine:

- keep both and say why (12.13 is the Tier-2 *presence* check, 13.1 is the pre-send *state* baseline — that's a defensible distinction, it just isn't written down); or
- do what the issue asked and drop §12.13 now that §13.1 covers it.

What shouldn't ship is a body claiming a removal that didn't happen, with `KY` still listed in a section header whose test may skip.

## Nits (non-blocking)

- If §12.13 stays, the two section strings at `:723`/`:728` are now conditionally accurate — `KY` is listed as a Tier-2 command but only exercised with `--cw`. Worth a parenthetical, e.g. `(… KY [--cw] …)`, so someone reading a skip-heavy run doesn't think a command silently vanished.
- The E6 comment is a good example of the house standard — it states the *mechanism* (`send_morse` → `CwxModel` → `"cwx"`) rather than restating the gate, which is what tells the next reader why it can't just be un-gated.

## Recommendation

**Approve with nits** — approval stays with @ten9876. Test-only, no production code, CI green on `build`/`check-windows`/`check-macos`, and the E6 gate closes a path where running the suite on a live radio would put CW on the air. That last point alone makes this worth landing promptly.

The fix I'd ask for before merge is one paragraph of the PR body, so the commit message records "gated" rather than "removed" — and a line either justifying or retiring the surviving `KY;` duplication.

*Verified by me:* that §12.13 is gated rather than removed and `KY` remains in both section strings; that §13.1 at `:904` queries `KY;` with the stricter assertion; that `doCw` is parsed at `:2278` and reaches `sectionEdge`. *Not run:* the suites themselves — no radio and no CAT port here, so the 89/89 and E6-skip results are read.

**Comment by K5PTB** [comment]
Thanks — all three addressed:

- **Body/issue accuracy:** PR body and #4855 now say §12.13 is *gated*, not removed, and note `KY` stays in the section headers. (Also dropped a stale `c5a6b95d` citation from the issue — that commit lives on `fix/cat-band-follow`, not `main`.)
- **Surviving duplication:** kept deliberately and now documented in both — §12.13 is the Tier-2 *presence* check (`KY0`/`KY1`), §13.1 is the pre-send *state* baseline (`== "KY0"`).
- **Nit (`ac2341d8`):** both Section-12 banner strings now read `KY [--cw]` so a skip-heavy demo run doesn't read as a command silently vanishing.
- **E6 comment:** noted, thanks — will hold new gate comments to that mechanism-over-restatement bar.

Test-only, still 89/89 with §12.13 `SKIP` on the demo. Ready when you are.

**Review by ten9876 — COMMENTED** [review]
## Re-review — the description/diff mismatch is fixed

My previous pass had no blockers and one accuracy complaint: the PR body said §12.13 was *removed* and `KY` *dropped from section 12's command list*, while the diff gated it and left `KY` in both the banner comment and the `r.section()` string. `ac2341d8` settles it the right way round — the code keeps the gating (which I said then and still think is the better call than deletion), and the two headers now read `KY [--cw]` so they describe what the section actually does:

```
// Section 12 — Tier-2 Commands (AG, GT, PC, NB, KS, PT, KY [--cw], RT, RG, RC, RD, RU, XT)
```

That is the accurate version of the claim, and annotating rather than removing keeps a Tier-2 section from silently losing a Tier-2 command — a reader of the transcript now sees `KY` listed and sees it skipped, with a reason, instead of not seeing it at all.

The new inline comment on the gate is also better than what I asked for. It draws the distinction that actually matters and that #4855 did not:

> The KY; *query* is read-only (it reports keyer-busy state, it never keys — that's `KY<text>;`), but the whole KY family is refused with `"?;"` on a rig without a radio-side keyer

That is the honest reason for the gate. The query is not a TX-safety risk; it is an availability problem. Saying so stops the next person from concluding that `KY;` keys something and "tightening" the gate elsewhere on that belief. The genuine Principle VI item is E6 in `rigctld_test.cpp`, where `send_morse` routes through `CwxModel` → `"cwx"` and does key CW — and that comment says exactly that too.

## Still open, and still not a blocker

The redundancy #4855 originally identified survives: under `--cw` on a keyered rig, `KY;` is queried twice — §12.13 (`CAT_TS-2000_test.cpp:817`, accepting `KY0` **or** `KY1`) and §13.1 (the stricter `== "KY0"` pre-send baseline). I said last time either resolution was fine, and gating both is a legitimate third answer. What is still missing is the one sentence saying the two are deliberately different questions — 12.13 is the Tier-2 *presence* check, 13.1 is the pre-send *state* baseline. Without it the next reader finds two queries of the same command four hundred lines apart and has to work out whether one is a leftover.

## One thing to check before merge

This PR reports three checks — `build`, `check-windows`, `check-macos` — but **not `Static checks`**, which the other PRs in flight all carry. It is a tests-only diff so a path filter is the likely explanation, but worth confirming it is a filter rather than a run that never fired, since branch protection is what has to be satisfied.

## Verdict

No blockers. Both halves of #4855 are now handled — the safety gate on E6 and the availability gate on §12.13 — and the headers no longer overstate what was done. The remaining item is a comment.

Verified: the `ac2341d8` diff and the resulting headers/gates at PR head, and the check list. Not run locally — these are integration tests that need a live CAT target.

**Comment by K5PTB** [comment]
Comment added in `b8c759a2` — you're right that my last reply put the presence-vs-baseline distinction in the PR body and the issue, not in the source where the next reader meets the two queries. §12.13's gate comment now names §13.1 explicitly and says which question each one asks, with a reciprocal pointer at §13.1 so a reader arriving from either direction gets the answer instead of having to work out whether one is a leftover:

```cpp
    // Section 13's 13.1 queries KY; too — deliberately, not as a leftover. They
    // ask different questions: 12.13 is the Tier-2 *presence* check (KY0 *or*
    // KY1 — the command answers at all), 13.1 is the pre-send *state* baseline
    // (strictly == "KY0" — the buffer is idle before section 13 keys).
```

Comments only — no assertion, gate, or test-count change.

On `Static checks`: it's a path filter, not a run that never fired. `static-checks.yml` triggers on `src/**`, `resources/themes/**`, `resources/shaders/**`, `CMakeLists.txt` and a specific `tools/*.py` list — there is no `tests/**` entry, so a tests-only diff correctly skips it. The workflow's own header says the same thing about protection:

> None of these were required status checks, so consolidating them does not touch branch protection (required contexts are build / check-windows / check-macos).

All three required contexts are green here.

That leaves the approval itself — the PR still reads `REVIEW_REQUIRED`, so it's merge-blocked on the click rather than on anything in the diff.

**Review by ten9876 — APPROVED** [review]
## Approving

Both halves of #4855 are handled: E6's `send_morse` is gated behind `--cw` because it genuinely keys CW through `CwxModel` (Principle VI), and §12.13's `KY;` is gated because a keyerless target answers `?;` — an availability problem, not a safety one. The new inline comment draws that distinction explicitly, which is better than what I asked for: it stops the next reader concluding that `KY;` keys something and "tightening" a gate elsewhere on that belief. `ac2341d8`'s header annotation (`KY [--cw]`) also makes the section banners describe what the section actually does, which is what my earlier accuracy complaint was about.

### One nit — please open a follow-up issue

Under `--cw` on a keyered rig, `KY;` is now queried twice — §12.13 (`CAT_TS-2000_test.cpp:817`, accepting `KY0` **or** `KY1`) and §13.1 (the stricter `== "KY0"` pre-send baseline). Gating both is a legitimate resolution of the redundancy #4855 raised, but nothing says the two are deliberately different questions: 12.13 is the Tier-2 *presence* check, 13.1 is the pre-send *state* baseline. Without that sentence the next reader finds the same command queried twice four hundred lines apart and has to work out whether one is a leftover.

### One thing to confirm before merge

This PR reports `build`, `check-windows` and `check-macos` but **not** `Static checks`, which the other PRs in flight all carry. A tests-only diff makes a path filter the likely explanation, but it is worth confirming that rather than a run that never fired — branch protection is what has to be satisfied.

**Review by ten9876 — COMMENTED** [review]
## Correction — my approval's follow-up ask is already done

`b8c759a2` landed between my review and my approval and I did not catch it before writing that body. **Disregard the request to open a follow-up issue** — the sentence I asked for is there, and it is on *both* sides rather than just one:

- `CAT_TS-2000_test.cpp:816-820` — "Section 13's 13.1 queries `KY;` too — deliberately, not as a leftover. They ask different questions: 12.13 is the Tier-2 *presence* check (`KY0` *or* `KY1` — the command answers at all), 13.1 is the pre-send *state* baseline (strictly `== \"KY0\"`…)"
- `:908-909` — the reciprocal note at 13.1 saying it is stricter on purpose.

Annotating both ends is the right call: whichever one a reader lands on first, they are told the other exists and why. That fully closes the redundancy #4855 raised.

Approval stands and covers this head.
